modules/apps/transfer/keeper/relay.go

```diff
@@ -244,7 +250,7 @@ func (k Keeper) OnRecvPacket(ctx sdk.Context, packet channeltypes.Packet, data t
 		defer func() {
 			telemetry.SetGaugeWithLabels(
 				[]string{"ibc", types.ModuleName, "packet", "receive"},
-				float32(data.Amount),
+				float32(transferAmount.Int64()),
```

Comment:
why is this a float32 of an int64?

Reply:
well I think float32 is the argument type, not sure why it is int64 (I used the existing code format)

It will be updated to be big int

Reply:
Did some more digging. float32 is the argument type for metrics, so not getting around that. The SDK handles this by checking if the amount is a Int64 before setting the value. I'm not sure why Int64 is used and Uint64 isn't, since I believe float32 goes up to somewhere around ~2^127

For consistency, I'm sticking with Int64 unless we think we should increase it to uint64?